week4/solutions/scrabble/tests/test_hand.py

```diff
@@ -0,0 +1,480 @@
+  def assert_stdout(self, expected_output, mock_stdout):
+    hand = Hand()
+    hand.set({'e':1, 'v':2, 'n':1, 'i':1, 'l':2})
+    hand.updated_hand = {'e':1, 'v':2, 'n':1, 'i':1, 'l':2}
+    hand.display()
+    
+    self.assertEqual(mock_stdout.getvalue().strip('\n'), expected_output)
+
+  def test(self):
+    self.assert_stdout('e v v n i l l')
+
+class HandDisplayTest2(unittest.TestCase):
+  @unittest.mock.patch('sys.stdout', new_callable=io.StringIO)
+  def assert_stdout(self, expected_output, mock_stdout):
+    hand = Hand()
+    hand.set({'e':1, 'v':2, 'n':1, 'i':1, 'l':2})
+    hand.updated_hand = {'e':4, 'v':2, 'y':1, 'i':1, 'o':3}
+    hand.display()
+    
+    self.assertEqual(mock_stdout.getvalue().strip('\n'), expected_output)
+    self.assertEqual(hand.get_hand_string(), 'e e e e v v y i o o o')
+
+  def test(self):
+    self.assert_stdout('e e e e v v y i o o o')
+
+
+# Hand.get_vowels()
+VOWEL_SET = set(['a', 'e', 'i', 'o', 'u'])
+def vowels_test(self, hand_size):
+  hand = Hand(hand_size)
+  hand.get_vowels()
+  number_vowels = sum(hand.get().values())
+  expected_number_vowels = hand_size//3
+  are_vowels = set(hand.get().keys()) <= VOWEL_SET
+  
+  self.assertEqual(number_vowels, expected_number_vowels)
+  self.assertEqual(are_vowels, True)
+
+class HandGetVowelsTest1(unittest.TestCase):
+  def test(self):
+    '''hand = Hand(6)
+    hand.get_vowels()
+    number_vowels = sum(hand.get()[key] for key in hand.get())
+    are_vowels = set(hand.get().keys()) <= VOWEL_SET
+    self.assertEqual(number_vowels, 2)
+    self.assertEqual((set(hand.get().keys()) <= VOWEL_SET), True)'''
+    vowels_test(self, 6)
+
+class HandGetVowelsTest2(unittest.TestCase):
+  def test(self):
+    '''hand = Hand(9)
+    hand.get_vowels()
+    number_vowels = sum(hand.get()[key] for key in hand.get())
+    are_vowels = set(hand.get().keys()) <= VOWEL_SET
+    self.assertEqual(number_vowels, 3)
+    self.assertEqual((set(hand.get().keys()) <= VOWEL_SET), True)'''
+    vowels_test(self, 9)
+
+class HandGetVowelsTest3(unittest.TestCase):
+  def test(self):
+    '''hand = Hand(30)
+    hand.get_vowels()
+    number_vowels = sum(hand.get()[key] for key in hand.get())
+    are_vowels = set(hand.get().keys()) <= VOWEL_SET
+    self.assertEqual(number_vowels, 10)
+    self.assertEqual((set(hand.get().keys()) <= VOWEL_SET), True)'''
+    vowels_test(self, 30)
+
+class HandGetVowelsTest4(unittest.TestCase):
+  def test(self):
+    '''hand = Hand(11)
+    hand.get_vowels()
+    number_vowels = sum(hand.get()[key] for key in hand.get())
+    are_vowels = set(hand.get().keys()) <= VOWEL_SET
+    self.assertEqual(number_vowels, 3)
+    self.assertEqual((set(hand.get().keys()) <= VOWEL_SET), True)'''
+    vowels_test(self, 11)
+
+
+# Hand.get_consonants()
+CONSONANT_SET = set(['b', 'c', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n',
+                  'p', 'q', 'r', 's', 't', 'v', 'w', 'x', 'y', 'z'])
+
+def consonant_test(self, hand_size):
```

Comment:
I ended up writing a function I could call in my tests for get_consonants and get_vowels that might clean up the set up a bit, but I wasn't sure if that is a good idea, so I let the full set up in the tests and commented it out.